docs/runtime-user-guide.md

```diff
@@ -0,0 +1,119 @@
+# Runtime User Guide
+
+This document is focused on clarifying the expectations of the user
+container running in a Knative environment. This is separate from the expectations of the
+environment itself. The expectations of the environment are detailed in the [runtime
+contract](https://github.com/knative/serving/blob/master/docs/runtime-contract.md).
+
+The target audience of this document is _developers_ and _language and tooling
+developers_ as defined below:
+
+- **Developers** write code which is packaged into a container which is run on
+  the Knative cluster.
+- **Language and tooling developers** typically write tools used by
+    _developers_ to package code into containers. As such, they are concerned
+    that tooling which wraps developer code complies with this runtime contract.
```

Comment:
Add a section clarifying that this is descriptive advice rather than prescriptive requirements? "Your container will fit better if it follows the following guidance..."